pkg/dataobj/internal/dataset/page.go

```diff
@@ -0,0 +1,104 @@
+package dataset
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+	"hash/crc32"
+	"io"
+
+	"github.com/golang/snappy"
+	"github.com/klauspost/compress/zstd"
+
+	"github.com/grafana/loki/v3/pkg/dataobj/internal/metadata/datasetmd"
+)
+
+// Helper types.
+type (
+	// PageData holds the raw data for a page.
+	PageData []byte
+
+	// PageInfo describes a page.
+	PageInfo struct {
+		UncompressedSize int    // UncompressedSize is the size of a page before compression.
+		CompressedSize   int    // CompressedSize is the size of a page after compression.
+		CRC32            uint32 // CRC32 checksum of the page after encoding and compression.
+		RowCount         int    // RowCount is the number of rows in the page, including NULLs.
+
+		Value       datasetmd.ValueType       // Type of values stored in the page.
+		Compression datasetmd.CompressionType // Compression used for values in the page.
```

Comment:
Why do we have this in page and columns ? Can they be different ?

Reply:
In my prototyping I had run into problems with trying to generically iterate over pages but not knowing what their encoding was, so value/compression/encoding got put on the page. 

I don't think that's necessary anymore (and if it is, we can bring it back). I'll put encoding type on the page metadata and value and compression on the column metadata.